ramlfications/parser/types.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2015 Spotify AB
+
+from __future__ import absolute_import, division, print_function
+import attr
+import copy
+from six import MAXSIZE
+from six import iteritems
+
+from ..errors import UnknownTypeError
+from ..parameters import Content
+
+__type_registry = {}
+
+def type_class(klass):
+    if klass.type is not None:
+        __type_registry[klass.type] = klass
+    return klass
+
+def create_type(name, raml_def):
+    # Work on a copy. We will pop the type. Don't modify the raw raml object.
+    raml_def = copy.copy(raml_def)
+
+    # spec:
+    # string is default type when nothing else defined
+    typeexpr = raml_def.pop('type', 'string')
```

Comment:
its not a deep copy, just a copy of the type the goal of this dict is just to be passed as keyword to the constructor of the type class.
as 'type' is not an attr.ib of the class then we need to pop it.
so that is a small copy of 5-10 references